query-coordinator/src/main/scala/com/socrata/querycoordinator/CompoundQueryRewriter.scala

```diff
@@ -0,0 +1,112 @@
+package com.socrata.querycoordinator
+
+import com.socrata.querycoordinator.QueryRewriter.{Anal, ColumnRef, Expr, RollupName}
+import com.socrata.soql.typed.ColumnRef
+import com.socrata.soql.{BinaryTree, Compound, Leaf, PipeQuery, SoQLAnalysis, typed}
+
+/**
+  * Rewrite compound query with either exact tree match or prefix tree match
+  * TODO: Make matching more flexible.  e.g. "SELECT a, b" does not match "SELECT b, a" or "SELECT a"
+  */
+trait CompoundQueryRewriter { this: QueryRewriter =>
+
+  /**
+    * match if the whole tree is equal or prefix is equal
```

Comment:
maybe we can get more specific here: "rewrite tree or prefix to match rollup if possible" (<- this is probably not actually accurate, I think you can explain better)